### PR TITLE
folding stocks

### PR DESCRIPTION
**Pull request by stanalbatross**
## About The Pull Request

adds folding stocks to m41 series guns, buffs solid stock a bit, resprited mk1

## Why It's Good For The Game

new stock = more stock love plus they get them in the movies

also will allow for easy transitioning between chaser/stationary fire style

I can consider adding a downside while folding so this is not a mindless pick for the gun

## Changelog

:cl: Stan_Albatross, Triiodine
add: added an M41 series folding stock, can be attached to the M41aMK1, M41aMK2, and M46c (as well as sister guns). Is slightly worse than the solid stock, but can be folded. Sprites by Triiodine
balance: buffed the rifle solid stock a bit
imageadd: resprited the MK1, by Triiodine
imageadd: added map variants for the m46c, by Triiodine
/:cl:

### CODE REVIEW

**Review by nauticall — CHANGES_REQUESTED** [review]
https://media.discordapp.net/attachments/571822035278495786/994549512561508422/unknown.png

All I have to say right now I've already said behind closed doors - that the M46C's stock looks rather jutting and more like a recoil spring than an actual stock holder. 

Besides that, me likey! Just that one thing.

File: code/modules/projectiles/gun_attachables.dm [hunk 1]
```diff
@@ -1236,17 +1241,84 @@ Defined in conflicts.dm of the #defines folder.
...
+	collapse_delay = 0.5 SECONDS
+	flags_attach_features = ATTACH_REMOVABLE|ATTACH_ACTIVATION
+	attachment_action_type = /datum/action/item_action/toggle
+
+/obj/item/attachable/stock/rifle/collapsible/New()
+	..()
+	//it makes stuff better when two-handed
+	accuracy_mod = HIT_ACCURACY_MULT_TIER_2
+	recoil_mod = -RECOIL_AMOUNT_TIER_5
+	scatter_mod = -SCATTER_AMOUNT_TIER_9
```
Comment (Nanu308) [inline code/modules/projectiles/gun_attachables.dm:1278]: Why is the collapsible stock better than normal, i thought this was meant to be worse?

**Review by Nanu308 — CHANGES_REQUESTED** [review]

File: code/modules/projectiles/gun_attachables.dm [hunk 2]
```diff
@@ -1236,17 +1241,84 @@ Defined in conflicts.dm of the #defines folder.
...
+	name = "\improper M41A folding stock"
+	desc = "The standard back end of any gun starting with \"M41\". Compatible with the M41A series, this stock reduces recoil and improves accuracy, but at a reduction to handling and agility. Also enhances the thwacking of things with the stock-end of the rifle."
+	slot = "stock"
+	melee_mod = 5
+	size_mod = 1
+	icon_state = "m41_folding"
+	attach_icon = "m41_folding_a"
+	pixel_shift_x = 40
+	pixel_shift_y = 14
+	wield_delay_mod = WIELD_DELAY_FAST
```
[the remark this answers is not in the document]
Comment (stanalbatross) [reply code/modules/projectiles/gun_attachables.dm:1265]: it only makes wielding faster when negative
when positive it makes wielding slower

File: code/modules/projectiles/gun_attachables.dm (hunk 1 above)
Comment (stanalbatross) [reply code/modules/projectiles/gun_attachables.dm]: this is worse :: accuracy mod is 1 tier lower, recoil and scatter mods are reduced by less

**Comment by Nanu308** [comment]
Can be tuned down the line further, good starting point tho.

**Comment by stanalbatross** [comment]
> image
> 
> All I have to say right now I've already said behind closed doors - that the M46C's stock looks rather jutting and more like a recoil spring than an actual stock holder.
> 
> Besides that, me likey! Just that one thing.

image

changed by trii

**Comment by Agameofscones** [comment]
I am approving my own sprites

**Review by morrowwolf — APPROVED** [review]
LGTM

**Pull request merged**